lib/board.rb

```diff
@@ -34,8 +34,48 @@ def free_positions
     board.flatten.select { |cell| cell.is_a?(Integer) }
   end
 
+  def string_to_board(board_string)
+    rows = board_string.split(",")
+
+    board.size.times do |index_row|
+      board.size.times do |index_column|
+        if rows[index_row].chars[index_column] == Mark::CROSS
+          @counter += 1
+          board[index_row][index_column] = Mark::CROSS
+        elsif rows[index_row].chars[index_column] == Mark::ROUND
+          @counter += 1
+          board[index_row][index_column] = Mark::ROUND
+        end
+      end
+    end
+  end
+
+  def to_s
```

Comment:
I advocate against using `to_s` for business specific behaviour.
It's preferred to have a specific method for that as `to_s` (similar to Java's `toString()` method) should not have domain specific knowledge. I always think of `to_s` and `toString()` as methods that might help me log a certain object, etc.

Additionally to that the behaviour this method implements is really a UI concern. Why should the `Board` itself know about the details about where to put spaces and commas?